File: Chap5/Scala/Parity.scala
// Actually I don't know how to write a good scala program.
// yet.
object Parity {
  def parity1(x: Int): Int = {
    if (x == 0) return 0
    return (x & 1) ^ parity1(x >> 1)
  }
  
  def main(args: Array[String]) {
    println(parity1(9))
  }
}

File: Chap5/Java/parity.java
class parity {
	static short parity1(int x) {
		short result = 0;

		while (x != 0) {
			result ^= (x & 1);
			x >>= 1;
		}

		return result;
	}

	public static void main(String[] args) {
		System.out.println("parity1(8)=" + parity1(8));
		System.out.println("parity1(9)=" + parity1(9));
	}
}
